src/components/Experiments/ExperimentEditModal.js

```diff
@@ -0,0 +1,182 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import {
+  Button,
+  Modal,
+  FormGroup,
+  FormControl,
+  Glyphicon,
+  Row,
+  Col,
+} from 'react-bootstrap';
+
+import {
+  create as experimentCreate,
+  update as experimentUpdate,
+  uploadFilePairs,
+  UPLOAD_RES_SUCCESS,
+  UPLOAD_RES_FAILURE,
+} from '../../state/experiments';
+
+import './ExperimentEditModal.less';
+
+class ExperimentEditModal extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = this.buildState(this.props);
+
+    this.handleNameChange = this.handleNameChange.bind(this);
+    this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
+    this.handleSave = this.handleSave.bind(this);
+    this.handleUpload = this.handleUpload.bind(this);
+  }
+
+  buildState(props) {
+    const exp = props.editModalExperiment;
+
+    if (props.action === 'edit' && exp !== undefined) {
+      return { nameVal: exp.name, descriptionVal: exp.description };
+    }
+
+    return { nameVal: '', descriptionVal: '' };
+  }
+
+  handleNameChange(e) {
+    this.setState({ nameVal: e.target.value });
+  }
+
+  handleDescriptionChange(e) {
+    this.setState({ descriptionVal: e.target.value });
+  }
+
+  handleSave() {
+    const create = this.props.action === 'create';
+
+    if (create) {
+      return this.props
+        .experimentCreate(this.state.nameVal, this.state.descriptionVal)
+        .then(expId => {
+          this.props.onOpenEdit(expId);
```

Comment:
It is kind of weird that after creating an experiment, the same modal is opened again without giving any special feedback to the user.

After creating a new experiment, I wouldn't open the modal automatically (as in edition-mode), so the user should open it manually -> she/he will be then aware of the edit-state.